tools/api-client/python/lib/random_ai.py

```diff
@@ -416,17 +416,30 @@ def _write_entry_type_submitTurn(self, entry):
       entry['attacker_indices'], entry['defender_indices'],
       entry['all_attackers'], entry['all_defenders'],
       entry['attacker'], entry['defender'])
-    php_turbo_vals = 'array(%s)' % ', '.join(['%s => %s' % (k, v) for k, v in sorted(entry['turbo_vals'].items())])
-    self.f.write("""
-        $this->verify_api_submitTurn(
-            %s,
-            '%s',
-            $retval, %s,
-            $gameId, %s, '%s', %d, %d, '', %s);
+    if entry['retval'].status == 'ok':
+      php_turbo_vals = 'array(%s)' % ', '.join(['%s => %s' % (k, v) for k, v in sorted(entry['turbo_vals'].items())])
+      self.f.write("""
+          $this->verify_api_submitTurn(
+              %s,
+              '%s',
+              $retval, %s,
+              $gameId, %s, '%s', %d, %d, '', %s);
 """ % (
-      randstr, self._php_get_message(entry['retval']), php_attack_str,
-      entry['roundNumber'], entry['attackType'], entry['attacker'], entry['defender'],
-      php_turbo_vals))
+        randstr, self._php_get_message(entry['retval']), php_attack_str,
+        entry['roundNumber'], entry['attackType'], entry['attacker'], entry['defender'],
+        php_turbo_vals))
+    elif entry['retval'].status == 'failed':
+      self.f.write("""
+          $this->verify_api_submitTurn_failure(
+              %s,
+              '%s',
+              $retval, %s,
+              $gameId, %s, '%s', %d, %d, '');
```

Comment:
I'm not offhand sure *why* `verify_api_submitTurn()` takes a `turboVals` arg and `verify_api_submitTurn_failure()` doesn't, but that is the state of play in responderTest, and this code's job is to match what responderTest expects, so that's what i did.

Reply:
I've taken a look and verify_api_submitTurn*() are only used in the responder tests, so I think this doesn't matter much in the greater scheme of things.